nmrfx-analyst/src/main/java/org/nmrfx/processor/datasets/vendor/NMRDataUtil.java

```diff
@@ -387,42 +405,129 @@ public static String getProcessedDataset(File localFile) {
                                 }
                             }
                     );
-                    datasetName = processed.get(0).getFileName().toString();
                 }
             }
         } catch (IOException ex) {
             log.warn(ex.getMessage(), ex);
         }
-        return datasetName;
-
+        return processed;
     }
 
     public static List<DatasetSummary> scanDirectory(String scanDir, Path savePath) {
```

Comment:
It is very tempting to remove this from the utility class and move it closer to the dataset browser, since that's the only usage.

Reply:
Moved scanDirectory and getProcessedDataset to datasetbrowser directory in file DatasetBrowserUtils